src/Symfony/Component/Serializer/Normalizer/DateTimeZoneNormalizer.php

```diff
@@ -20,8 +20,15 @@
  *
  * @author Jérôme Desjardins <jewome62@gmail.com>
  */
-class DateTimeZoneNormalizer implements NormalizerInterface, DenormalizerInterface, CacheableSupportsMethodInterface
+class DateTimeZoneNormalizer implements NormalizerInterface, DenormalizerInterface, CacheableSupportsMethodInterface, SupportedTypesMethodInterface
 {
+    public function getSupportedTypes(): ?array
+    {
+        return [
+            \DateTimeZone::class => $this->hasCacheableSupportsMethod(),
```

Comment:
Looking at the concrete normalizers in this PR already, it seems there will be a lot of overlap of `getSupportedTypes` and `supportsNormalization`. In most cases they tell the same information.

Perhaps `supportsNormalization` can become optional for normalizers that suffice with `getSupportedTypes` (like this `DateTimeZoneNormalizer `). Or a trait with

```
    public function supportsNormalization(mixed $data, string $format = null /* , array $context = [] */): bool
    {
        foreach ($this->getSupportedTypes() as $type => $supported) {
            if ($data instanceof $type) {
                return true;
            }
       }

       return false;
    }
```

Reply:
I like this as well.

But then, as the Serializer already checks the type, does it makes sense to check it here the exact same way? looks like unnecessary logic duplication 🤔

Question: would that makes sense, in that case, to also allows (de)normalizers to expose their supported formats and let only the normalizers requiring it expose a `supportsNormalizationContext` method (or similar) when they need more fine-grained control?
It looks to me like it could work well with @dunglas idea to make `supportsNormalization` optional.
Cacheability would not be an issue anymore as we would know which normalizers depend on the context 
WDYT?

Reply:
no need to make `supportsNormalization` optional: we can just make it return true in the base class